core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.cpp

```diff
@@ -127,7 +128,7 @@ void OpenMPTargetExec::resize_scratch(int64_t team_size, int64_t shmem_size_L0,
   // on the maximum number of in-flight teams possible.
   int64_t total_size =
       (shmem_size + OpenMPTargetExecTeamMember::TEAM_REDUCE_SIZE + padding) *
-      (MAX_ACTIVE_THREADS / team_size);
+      std::min(MAX_ACTIVE_THREADS / team_size, league_size);
```

Comment:
This makes sense but if we have two consecutive kernels requesting scratch memory where `kernel2` follows `kernel1` and `MAX_ACTIVE_THREADS/team_size > kernel2.league_size() > kernel1.league_size()`, we end up allocating memory twice instead of just once. Otherwise this looks good.

Reply:
Yes. Do you suggest doing something else? Note that I only introduced it for #4728 do pass which is requesting a lot of memory with a low number of teams.

Reply:
The request in https://github.com/kokkos/kokkos/issues/4143 was to avoid excessive memory allocations even if they are still fitting on the device and thus the current behavior in `OMPT` goes against this.
The memory we allocate is per execution space instance. Hence, the scenario you are describing only really matters if you only run these two kernels, right? If they are run again, no more memory would be allocated anyway.
Interestingly, `scratch_size_max` isn't used internally at all (in any backend).

In the end, I think we should find behavior that works similarly across backends. Maybe, someone else can weigh in here to decide where we want to go.

Reply:
I am not adamant about the change and I think it is probably more beneficial to allocate just the `league_size` amount of scratch memory as in this PR. I was just trying to think of an alternative case.

Reply:
I am also not adamant about this. :slightly_smiling_face: